src/dist/dist.rs

```diff
@@ -642,6 +650,10 @@ fn update_from_dist_<'a>(
                 }
 
                 if let ErrorKind::RequestedComponentsUnavailable(components, ..) = e.kind() {
+                    if fetched.is_empty() {
+                        // We failed to fetch, perhaps because the components are bad, stop now.
```

Comment:
I'm confused as to how this check is correct. `fetched` is set before `manifestation.update` is even called, so checking whether it is empty can't actually check whether components were available. It should only be empty if the entire manifest failed to download, but that sounds like a case where we *do* want to continue iterating backwards?

Reply:
I *think* this might be a bogon from a fiddle I was doing, I'll check, but I think it'll never fire.